src/components/Pagination.js

```diff
@@ -0,0 +1,69 @@
+import React from 'react';
+import { withRouter } from 'next/router';
+import { Link } from '../server/pages';
+
+import { Flex } from 'grid-styled';
+import StyledLink from './StyledLink';
+import { TextInput } from './StyledInput';
+import { Span } from './Text';
+
+const Pagination = ({ router, limit, offset, total }) => {
+  const { pathname, query, route } = router;
+  const totalPages = Math.ceil(total / limit);
+  const currentPage = offset / limit + 1;
+
+  const changePage = ({ target, key }) => {
+    if (key && key !== 'Enter') return;
+
+    const { value } = target;
+    if (!value) return;
+
+    router.push({ pathname, query: { ...query, offset: (value - 1) * limit } });
+  };
+
+  return (
+    <Flex alignItems="center">
+      {currentPage !== 1 && (
+        <Link
+          route={route.slice(1)}
+          scroll={false}
+          params={{ ...query, offset: offset - limit }}
+          passHref
+        >
+          <StyledLink buttonStyle="standard" buttonSize="small">
+            Previous
+          </StyledLink>
+        </Link>
+      )}
+      <Flex alignItems="center" mx={2}>
+        <Span>Page </Span>
+        <TextInput
+          key={offset}
+          defaultValue={currentPage}
+          onBlur={changePage}
+          onKeyPress={changePage}
+          textAlign="center"
+          mx={1}
+          px={1}
+          py={1}
+          width={30}
+        />
+        <Span>of {totalPages}</Span>
+      </Flex>
+      {currentPage !== totalPages && (
```

Comment:
I would use an 'less than' comparator here:
```suggestion
      {currentPage < totalPages && (
```

If for any reason (future update in limit, incorrect input in page field) the user ends up after the limit we don't want to show the next button:
![selection_757](https://user-images.githubusercontent.com/1556356/48889181-3bdfcc00-ee35-11e8-9452-820cc96f6ed0.png)

Reply:
Good suggestion. Will implement :+1: